mimic/model/heat_objects.py

```diff
@@ -9,6 +9,27 @@
 from mimic.model.behaviors import BehaviorRegistryCollection, EventDescription
 
 
+def get_stack_or_404(f):
+    """
+    Decorator which retrieves a stack from a collection and adds it to the
+    function's arguments if it exists. If the stack does not exist, a 404
```

Comment:
This is slightly misleading, since it modifies the signature and doesn't document the modified signature.

`do_something_behavior` takes `collection, request, stack`, and it says It "retrieves a stack from a collection and adds it to the function's arguments".  but it doesn't say that it *removes* the stack name and stack ID from the argument signature.

I generally don't like decorators which mangle function signatures this way, because they break one of the more basic ways that Python programmers read code: you look at a function and you call it with the arguments you see in the argument list.  I think this might make sense as a utility function.